java/source/conf.py

```diff
@@ -36,7 +36,12 @@
 project = 'Apache Arrow Java Cookbook'
 copyright = '2022, Apache Software Foundation'
 author = 'The Apache Software Foundation'
-
+arrow_nightly=os.getenv("ARROW_NIGHTLY")
+if arrow_nightly and arrow_nightly != '0':
+    version = "10.0.0-SNAPSHOT"
```

Comment:
Another option to register `version` could be at `yml` side as a job parameter. In some cases for example will be needed run the job with custom version without change the code or PR needed

Reply:
@davisusanibar do you think this comment still applies with the new versioning `10.0.0-SNAPSHOT`, I suspect that if we want to test a specific version for a specific date instead of using the repository we probably have to manually download the artifacts for that specific day but this seems a pretty corner case scenario and I am not sure over-engineering for it is worth it.

Reply:
Not problem how it is now, thank you